lib/rubygems/test_case.rb

```diff
@@ -256,6 +256,14 @@ def setup
       @tempdir.untaint
     end
 
+    # This makes the tempdir consistent on Windows.
+    # Dir.tmpdir may return short path name, but Dir[Dir.tmpdir] returns long
+    # path name. https://bugs.ruby-lang.org/issues/10819
+    if win_platform?
+      @tempdir = Dir[@tempdir][0]
```

Comment:
Short name to long name path conversion is from FindFirstFile() API function on Windows.
This is only implemented with `Dir.glob` (https://bugs.ruby-lang.org/issues/10819). Not with `File.expand_path` and `File.realpath`.
Short name and long name path is specific to Windows file system.

```
C:\Users\h.shirosaki>irb
irb(main):001:0> RUBY_DESCRIPTION
=> "ruby 2.3.0p0 (2015-12-25 revision 53290) [x64-mswin64_100]"
irb(main):002:0> require 'tmpdir'
=> true
irb(main):003:0> tempdir = Dir.tmpdir
=> "C:/Users/H3BDC~1.SHI/AppData/Local/Temp"
irb(main):004:0> Dir[tempdir]
=> ["C:/Users/h.shirosaki/AppData/Local/Temp"]
irb(main):005:0> File.expand_path tempdir
=> "C:/Users/H3BDC~1.SHI/AppData/Local/Temp"
irb(main):006:0> File.realpath tempdir
=> "C:/Users/H3BDC~1.SHI/AppData/Local/Temp"
```